File: ecs-java-s3-workshop/src/main/java/com/emc/ecs/s3/sample/_01_CreateObject.java
/*
 * Copyright 2013 EMC Corporation. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0.txt
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
package com.emc.ecs.s3.sample;

import com.emc.object.s3.S3Client;

import java.io.BufferedReader;
import java.io.InputStreamReader;

public class _01_CreateObject {

	public static void main(String[] args) throws Exception {
    	// create the ECS S3 Client
    	S3Client s3 = ECSS3Factory.getS3Client();

    	// retrieve object key/value from user
        System.out.println( "Enter the object key:" );
        String key = new BufferedReader( new InputStreamReader( System.in ) ).readLine();
        System.out.println( "Enter the object content:" );
        String content = new BufferedReader( new InputStreamReader( System.in ) ).readLine();
        
        // create the object in the demo bucket
        s3.putObject(ECSS3Factory.S3_BUCKET, key, content.getBytes("UTF-8"), null);

        // print bucket key/value and content for validation
    	System.out.println( String.format("created object [%s/%s] with content: [%s]",
    			ECSS3Factory.S3_BUCKET, key, content));
    }
}

File: ecs-net-workshop/aws-net-workshop/examples/_01_CreateObject.cs
using System;
using ECSSDK.S3;
using ECSSDK.S3.Model;

namespace ecs_net_workshop.examples
{
    class _01_CreateObject
    {
        public static void Main(string[] args)
        {
            // create the ECS S3 client
            ECSS3Client s3 = ECSS3Factory.getS3Client();

            // retrieve the object key and object content from the user
            Console.Write("Enter the object key: ");
            string key = Console.ReadLine();
            Console.Write("Enter the object content: ");
            string content = Console.ReadLine();

            // create object request with retrieved input
            PutObjectRequestECS request = new PutObjectRequestECS()
            {
                BucketName = ECSS3Factory.S3_BUCKET,
                ContentBody = content,
                Key = key
            };

            // create the object in demo bucket
            s3.PutObject(request);           

            // print out object key/value for validation
            Console.WriteLine(string.Format("Created object {0}/{1} with content: {2}", ECSS3Factory.S3_BUCKET, key, content));
            Console.ReadLine();
        }
    }
}